py/server/tests/test_data_index.py

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+#
+
+import unittest
+
+from deephaven import empty_table
+from deephaven.experimental.data_index import create_data_index, has_data_index, get_data_index
+from tests.testbase import BaseTestCase
+
+class DataIndexTestCase(BaseTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.table = empty_table(10).update(["Timestamp = now()", "X=i%3", "Y=`Deephaven` + String.valueOf(ii)", "Z=ii*2"])
+        self.data_index = create_data_index(self.table, ["X", "Y"])
+        self.table_nodi = empty_table(10).update(["Timestamp = now()", "X=i%3", "Y=`Deephaven` + String.valueOf(ii)",
+                                           "Z=ii*2"])
+
+    def test_has_data_index(self):
+        self.assertTrue(has_data_index(self.table, ["X", "Y"]))
+        self.assertFalse(has_data_index(self.table, ["X"]))
+        self.assertFalse(has_data_index(self.table, ["X", "Z"]))
+        self.assertFalse(has_data_index(self.table_nodi, ["X", "Y"]))
+        self.assertFalse(has_data_index(self.table_nodi, ["X"]))
```

Comment:
```suggestion
        self.assertFalse(has_data_index(self.table_nodi, ["X"]))
        self.assertFalse(has_data_index(self.table_nodi, ["X", "Z"]))
```
Seems like this should be here for symmetry.